core/src/main/java/org/pentaho/di/metastore/MetaStoreConst.java

```diff
@@ -53,6 +60,27 @@ public class MetaStoreConst {
 
   public static final String DB_ATTR_ID_ATTRIBUTES = "attributes";
 
+  private static final Supplier<IMetaStore> metaStoreSupplier = new Supplier<IMetaStore>() {
+    private volatile MetastoreLocatorOsgi metastoreLocator;
+
+    public IMetaStore get() {
+      if ( metastoreLocator == null ) {
+        synchronized ( this ) {
+          try {
+            if ( metastoreLocator == null ) {
+              Collection<MetastoreLocatorOsgi> metastoreLocators =
+                  PluginServiceLoader.loadServices( MetastoreLocatorOsgi.class );
+              metastoreLocator = metastoreLocators.stream().findFirst().orElse( null );
```

Comment:
When we pick the first one, how do we know which metastoreLocator it is going to return

Reply:
There should generally be only one.

Reply:
nvm, I was thinking about MetastoreProvider

Reply:
We should try and use the MetastoreLocator instead of MetastoreLocatorOsgi as MetastoreLocatorOsgi  is meant to be used when working with karaf

Reply:
Using MetastoreLocator would require kettle core to depend on the plugin... I don't think we want that.

Reply:
There is separate api jar - pentaho-metastore-locator-api